drivers/sensor/max17055/max17055.c

```diff
@@ -119,7 +119,7 @@ static int capacity_to_max17055(unsigned int rsense_mohms, uint16_t val_mha)
  */
 static int voltage_mV_to_max17055(uint16_t val_mv)
 {
-	return (val_mv * 16) * 10 / 8; /* * 1.25 */
+	return ((uint32_t)val_mv * 16) * 8 / 10; /* / 1.25 */
```

Comment:
This doesn't look right either. This function is being used to set the VEmpty register, which is composed of two fields, VE (empty voltage target) and VR (recovery voltage), which have 10 mV and 40 mV resolution respectively. They're also different widths, 9 bits and 7 bits. I don't see how the 16-bit `max17055_reg_write` function can set VE correctly without corrupting VR.

Reply:
Hey, you are right, this code was clearing VR the whole time! Thanks for spotting this, I've reworked the code to mask out the VR bit and leave it alone. I think that the driver could use some more care, reworking few bits definitions, renaming some fields to add the unit of measure etc, but for the moment I'd say let's fix this one as the driver is currently totally broken.

Reply:
Agreed, thanks